tests/js/open-behavior-tests.js

```diff
@@ -16,463 +16,567 @@
+
+        TestCase.assertFalse(Realm.exists(config));
+
+        const realm = await Realm.open(config);
+
+        TestCase.assertDefined(realm.path);
+
+        realm.close();
+        await user.logOut();
     },
 
-    testNewFile_downloadBeforeOpen: function() {
-        return getLoggedInUser()
-            .then(user => {
-                const config = user.createConfiguration({
-                    sync: {
-                        _sessionStopPolicy: 'immediately',
-                        newRealmFileBehavior: {
-                            type: 'downloadBeforeOpen'
-                        },
-                        url: 'realm://127.0.0.1:9080/new_realm_' + Utils.uuid()
-                    }
-                });
-                return Realm.open(config);
-            })
-            .then(realm => {
-                TestCase.assertTrue(realm.empty)
-                realm.close();
+    testExistingFile_openLocal: async function() {
+        // NOTE: this test no longer runs with a logged out user.
+        // Reason: Error: User is no longer valid.
+
+        const app = new Realm.App(APP_CONFIG);
+        const user = await app.logIn(Realm.Credentials.anonymous());
+        const partitionValue = Utils.genPartition();
+
+        {
+            const config = {
+                schema: [schemas.DogForSync],
+                sync: {
+                    user,
+                    partitionValue,
+                    newRealmFileBehavior: Realm.App.Sync.openLocalRealmBehavior
+                }
+            };
+
+            TestCase.assertFalse(Realm.exists(config));
+
+            const realm = new Realm(config);
+            realm.write(() => {
+                realm.create(schemas.DogForSync.name, { _id: new ObjectId(), name: "Bella" });
             });
+            realm.close();
+        }
+
+        {
+            const config = {
+                schema: [schemas.DogForSync],
+                sync: {
+                    user,
+                    partitionValue,
+                    existingRealmFileBehavior: { type: "openImmediately" }
+                }
+            };
+
+            const realm = await Realm.open(config);
+
+            TestCase.assertEqual(realm.objects(schemas.DogForSync.name).length, 1);
+
+            realm.close();
+        }
+
+        await user.logOut();
+    },
+
+    testNewFile_downloadBeforeOpen: async function() {
+        const app = new Realm.App(APP_CONFIG);
+        const user = await app.logIn(Realm.Credentials.anonymous());
+        const partitionValue = Utils.genPartition();
+
+        const config = {
+            schema: [schemas.DogForSync],
+            sync: {
+                user,
+                partitionValue,
+                _sessionStopPolicy: "immediately",
+                newRealmFileBehavior: { type: "downloadBeforeOpen" }
+            }
+        };
+
+        const realm = await Realm.open(config);
+
+        // NOTE: Not quite sure what we're testing here?
```

Comment:
new file should have no objects. Maybe just remove this comment cause the test was written like that before. Its in a context with the next tests which is `existingFile`

Reply:
Can't be in the context of the next test, that's done with a different user (so wont be the same realm-file, and the next test ensures an existing file before testing - this is what confused me about this test).
I only left my comment in to get a clue what's going on (will remove before merge) :)